airgun/views/virtwho_configure.py

```diff
@@ -0,0 +1,150 @@
+    filtering_content = ConditionalSwitchableView(reference='filtering')
+    hypervisor_type = FilteredDropdown(id='foreman_virt_who_configure_config_hypervisor_type')
+    hypervisor_content = ConditionalSwitchableView(reference='hypervisor_type')
+    submit = Text('//input[@name="commit"]')
+
+    @hypervisor_content.register(
+        lambda hypervisor_type: hypervisor_type.endswith(
+            ('(esx)', '(rhevm)', '(hyperv)', '(xen)')
+        )
+    )
+    class HypervisorForm(View):
+        server = TextInput(id='foreman_virt_who_configure_config_hypervisor_server')
+        username = TextInput(id='foreman_virt_who_configure_config_hypervisor_username')
+        password = TextInput(id='foreman_virt_who_configure_config_hypervisor_password')
+
+    @hypervisor_content.register('libvirt')
+    class LibvirtForm(View):
+        server = TextInput(id='foreman_virt_who_configure_config_hypervisor_server')
+        username = TextInput(id='foreman_virt_who_configure_config_hypervisor_username')
+
+    @hypervisor_content.register('Container-native virtualization')
+    class KubevirtForm(View):
+        server = TextInput(id='foreman_virt_who_configure_config_hypervisor_server')
+        kubeconfig = TextInput(id='foreman_virt_who_configure_config_kubeconfig_path')
+
+    @filtering_content.register('Unlimited', default=True)
+    class FilterUnlimitedForm(View):
+        pass
+
+    @filtering_content.register('Whitelist')
+    class FilterWhitelistForm(View):
+        filter_hosts = TextInput(
+            id='foreman_virt_who_configure_config_whitelist'
+        )
+        filter_host_parents = TextInput(
+            id='foreman_virt_who_configure_config_filter_host_parents'
+        )
+
+    @filtering_content.register('Blacklist')
+    class FilterBlacklistForm(View):
+        exclude_hosts = TextInput(
+            id='foreman_virt_who_configure_config_blacklist'
+        )
+        exclude_host_parents = TextInput(
+            id='foreman_virt_who_configure_config_exclude_host_parents'
+        )
+
+    @property
+    def is_displayed(self):
+        breadcrumb_loaded = self.browser.wait_for_element(
+            self.breadcrumb, exception=False)
+        return (
+            breadcrumb_loaded
+            and self.breadcrumb.locations[0] == 'Satellite Virt Who Configure Configs'
+            and self.breadcrumb.read() == 'New Virt-who Config'
+        )
+
+
+class VirtwhoConfigureEditView(VirtwhoConfigureCreateView):
+
+    @property
+    def is_displayed(self):
+        breadcrumb_loaded = self.browser.wait_for_element(
+            self.breadcrumb, exception=False)
+        return (
+            breadcrumb_loaded
+            and self.breadcrumb.locations[0] == 'Configurations'
+            and self.breadcrumb.read().startswith('Edit ')
+        )
+
+
+class VirtwhoConfigureDetailsView(BaseLoggedInView):
+    breadcrumb = BreadCrumb()
+    edit = Text("//a[text()='Edit']")
+    delete = Text("//a[text()='Delete']")
```

Comment:
I see the javascript function text() used multiple times, are you facing issues finding the locator based on id or class?

Reply:
Finding these elements based on their text is common idiom in Airgun, too.
They don't have any id we could tie to, and classes are very non-specific:
![Screenshot from 2019-07-05 11-18-22](https://user-images.githubusercontent.com/12373754/60713121-ba96fe80-9f18-11e9-8951-0639e0e90e69.png)

Reply:
Ok NP

Reply:
There is no id for them, and the class is not the unique for them, `//a[text()='Edit']` can work well to find the locator
```
<a class="btn btn-default" href="/foreman_virt_who_configure/configs/43/edit">Edit</a>
<a data-confirm="Delete virt-who configuration ?" class="btn btn-default" rel="nofollow" data-method="delete" href="/foreman_virt_who_configure/configs/43">Delete</a>
```